server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/cics/utility/CICSCollectStatisticsSpOptionsCheckUtility.java

```diff
@@ -0,0 +1,124 @@
+                    put(CICSLexer.TSQUEUE, ErrorSeverity.WARNING);
+                    put(CICSLexer.VTAM, ErrorSeverity.WARNING);
+                }
+            };
+    public CICSCollectStatisticsSpOptionsCheckUtility(
+            DialectProcessingContext context, List<SyntaxError> errors) {
+        super(context, errors, DUPLICATE_CHECK_OPTIONS);
+    }
+
+    /**
+     * Entrypoint to check CICS COLLECT STATISTICS system command rule options
+     *
+     * @param ctx ParserRuleContext subclass containing options
+     * @param <E> A subclass of ParserRuleContext
+     */
+    public <E extends ParserRuleContext> void checkOptions(E ctx) {
+        if (ctx.getRuleIndex() == RULE_cics_collect_statistics_opts)
+            checkOpts((CICSParser.Cics_collect_statistics_optsContext) ctx);
+
+        checkDuplicates(ctx);
+    }
+    private void checkOpts(CICSParser.Cics_collect_statistics_optsContext ctx) {
+        checkHasMandatoryOptions(ctx.SET(), ctx, "SET");
+        checkHasMutuallyExclusiveOptions("LASTRESET or LASTRESETHRS", ctx.LASTRESET(), ctx.LASTRESETHRS());
+        if (!ctx.LASTRESETHRS().isEmpty() || !ctx.LASTRESETMIN().isEmpty() || !ctx.LASTRESETSEC().isEmpty()) {
+            checkHasMandatoryOptions(ctx.LASTRESETHRS(), ctx, "LASTRESETHRS");
+            checkHasMandatoryOptions(ctx.LASTRESETMIN(), ctx, "LASTRESETMIN");
+            checkHasMandatoryOptions(ctx.LASTRESETSEC(), ctx, "LASTRESETSEC");
+        }
+        checkHasMutuallyExclusiveOptions("AUTOINSTALL or  CONNECTION or  DB2CONN or "
+                        + "DB2ENTRY or  DISPATCHER or  ENQUEUE or  FILE or  JOURNALNAME or  JOURNALNUM or "
+                        + "JVMPROGRAM or  LSRPOOL or  MONITOR or  MVSTCB or "
+                        + "PROGAUTO or  PROGRAM or  RECOVERY or  STATS or  STORAGE or  STREAMNAME or "
+                        + "SUBPOOL or  SYSDUMPCODE or  TABLEMGR or  TARGET or  TASKSUBPOOL or "
+                        + "TCLASS or  TCPIP or  TCPIPSERVICE or  TDQUEUE or  TERMINAL or  TRANCLASS or "
+                        + "TRANDUMPCODE or  TRANSACTION or  TSQUEUE or  VTAM", ctx.AUTOINSTALL(), ctx.CONNECTION(), ctx.DB2CONN(),
+                ctx.DB2ENTRY(), ctx.DISPATCHER(), ctx.ENQUEUE(), ctx.FILE(), ctx.JOURNALNAME(), ctx.JOURNALNUM(),
+                ctx.JVMPROGRAM(), ctx.LSRPOOL(), ctx.MONITOR(), ctx.MVSTCB(),
+                ctx.PROGAUTO(), ctx.PROGRAM(), ctx.RECOVERY(), ctx.STATS(), ctx.STORAGE(), ctx.STREAMNAME(),
+                ctx.SUBPOOL(), ctx.SYSDUMPCODE(), ctx.TABLEMGR(), ctx.TARGET(), ctx.TASKSUBPOOL(),
+                ctx.TCLASS(), ctx.TCPIP(), ctx.TCPIPSERVICE(), ctx.TDQUEUE(), ctx.TERMINAL(), ctx.TRANCLASS(),
+                ctx.TRANDUMPCODE(), ctx.TRANSACTION(), ctx.TSQUEUE(), ctx.VTAM());
+        if (ctx.NODE().isEmpty()) checkPrerequisiteIsMet(ctx.POOL(), ctx.TARGET(), ctx, "TARGET without POOL");
+        if (ctx.POOL().isEmpty()) checkPrerequisiteIsMet(ctx.TARGET(), ctx.NODE(), ctx, "NODE without TARGET");
```

Comment:
all or nothing - if there is no POOL then either both TARGET and NODE are specified or neither.